### ISSUE TITLE
Missing way how to pass own env. variables or configuration

### ISSUE THREAD

**Issue by synaptiko** [fmoralesc/neovim-gnome-terminal-wrapper#8]
Hi Felipe,

I appreciate your work on this wrapper. I find out that following could be useful improvements:
- default configuration file
- config parameter when running nvim-wrapper (so you could have custom .desktop file with some parameters overriden)

Particulary I would like to configure:
- gnome-terminal params (e.g. --maximized, --full-screen)
- environment variables, either disable default ones (NVIM_*) or enable new ones (FZF_DEFAULT_COMMAND)
- or maybe better way, if ~/.bashrc should be sourced before gnome-terminal/nvim is run

What do you think?

I'll try to prepare pull-request if I find some free time. Currently I've just updated nvim-wrapper directly according to my needs and installed it as different package through makepkg/pacman.

**Comment by fmoralesc** [comment]
I would be OK with merging any of this functionality. However,

1) Instead of introducing configuration parameters or configuration files, we might want to simply pass through arguments to gnome-terminal.
2) If you are willing to use a custom .desktop file, then it is already possible to set up environment variables using env:

```
Exec=env NVIM_X=x nvim-wrapper %
```

3) Sourcing initialization files might not be useful in every situation. For example, I use fish instead of bash. Adding support for several shells just introduces complexity.

OTOH, we could install a new gnome-terminal profile and allow the user to configure it.

So yeah, I think this is a good idea, but I hope we could introduce this in a way that is stupid simple.

**Comment by synaptiko** [comment]
I would like to have:
A) Neovim wrapper as generic text editor, so when I click on a file in GUI, the file will be opened in Neovim
B) Neovim wrapper with specific configuration per project

Current defaults are ok for me, I only want to override NVIM_TUI_\* variables for both A and B and there is no way how to do it.

Then for B I also want to start gnome-terminal in full-screen and set current-directory. For Case B I could optionally want to create custom desktop file.

So, I agree with you that it should be stupid simple. Definitely, 1 and 2 are ok for Case B. Also If there is a possibility to use profile for 3, I would prefer it too (it's probably "Run a custom command..." option, isn't it?). But still there is no way how to override env. variables which are set inside nvim-wrapper.

To summarize it, I think it would be good to either not set env. variables inside the nvim-wrapper or set them only in case they are not already present (NVIM_TUI_\* with empty value to be able to disable it). Also adding support for passing arguments over nvim-wrapper to gnome-terminal would help, then I could call `nvim-wrapper --full-screen --current-directory=~/whatever`.

**Comment by synaptiko** [comment]
Now it's possible to run:
`NVIM_TUI_ENABLE_CURSOR_SHAPE= nvim-wrapper --full-screen -working-directory=~/work myFile.js`

Thanks for assistence.

**Comment by fmoralesc** [comment]
I did nothing, really. Thank you.
